bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -2596,6 +2596,12 @@ public void setExpandedState(Object elementOrTreePath, boolean expanded) {
 		Assert.isNotNull(elementOrTreePath);
 		if (checkBusy())
 			return;
+
+		// we do not support expand/collapse for ExpandableNode
+		if (elementOrTreePath instanceof ExpandableNode) {
```

Comment:
Why not `isExpandableNode(elementOrTreePath)` ?

Reply:
yes